ical_exporter_test.go

```diff
@@ -0,0 +1,92 @@
+package main
+
+import (
+	"testing"
+	"bytes"
+)
+
+func TestICalExport(t *testing.T) {
+	tasks := Tasks{
+		Items: []Task{
+			{"identifier-1", TaskStart, "2016-01-02T15:04:00Z"},
+			{"identifier-1", TaskStop, "2016-01-02T17:04:02Z"},
+			
+			{"identifier-2", TaskStart, "2016-01-02T15:04:00Z"},
+			{"identifier-2", TaskStop, "2016-01-02T16:04:00Z"},
+
+			{"identifier-1", TaskStart, "2017-01-01T19:06:02Z"},
+			{"identifier-1", TaskStop, "2017-01-01T19:06:03Z"},
+		},
+	}
+	buffer := bytes.Buffer{}
+	exporter := ICalTaskExporter{}
+
+	err := exporter.Export(tasks, &buffer)
+
+	if err != nil {
+		t.Error("Failed export.")
+	}
+
+	expectedResult := `BEGIN:VCALENDAR
```

Comment:
would put this into a fixtures file with constants or in the same file, but not in the body of the function

Reply:
I am a fan of placing text near the comparison in tests unless it is a big CLOB. But this is more of habit i have when debugging in an IDE so i can quickly check the differences. But if i used a testing lib function that highlights the differences this would lose value.

I ll move them to constants in the top of the file.